masq_lib/src/percentage.rs

```diff
@@ -227,91 +141,271 @@ impl PurePercentage {
+// The expected use of this would look like requesting percents meaning possibly multiples of 100%,
+// roughly, of a certain base number. Similarly to the PurePercentage type, also signed numbers
+// would be accepted.
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct LoosePercentage {
+    multiplier_of_100_percent: u32,
+    degrees_from_remainder: PurePercentage,
+}
+
+impl LoosePercentage {
+    pub fn new(percents: u32) -> Self {
+        let multiples_of_100_percent = percents / 100;
+        let remainder = (percents % 100) as u8;
+        let degrees_from_remainder =
+            PurePercentage::try_from(remainder).expect("Should never happen.");
+        Self {
+            multiplier_of_100_percent: multiples_of_100_percent,
+            degrees_from_remainder,
+        }
+    }
+
+    // If this returns an overflow error, you may want to precede this by converting the base
+    // number to a larger integer
+    pub fn of<N>(&self, num: N) -> Result<N, BaseTypeOverflow>
+    where
+        N: PercentageInteger + TryFrom<u32>,
+        <N as TryFrom<i8>>::Error: Debug,
+        <N as TryFrom<u32>>::Error: Debug,
+        i16: TryFrom<N>,
+        <i16 as TryFrom<N>>::Error: Debug,
+    {
+        let multiplier = match N::try_from(self.multiplier_of_100_percent) {
+            Ok(n) => n,
+            Err(e) => {
+                return Err(BaseTypeOverflow {
+                    msg: format!(
+                        "Couldn't init multiplier {} to type {} due to {:?}.",
+                        self.multiplier_of_100_percent,
+                        type_name::<N>(),
+                        e
+                    ),
+                })
+            }
+        };
+
+        let wholes = match num.checked_mul(&multiplier) {
+            Some(n) => n,
+            None => {
+                return Err(BaseTypeOverflow {
+                    msg: format!(
+                        "Multiplication failed between {:?} and {:?} for type {}.",
+                        num,
+                        multiplier,
+                        type_name::<N>()
+                    ),
+                })
+            }
+        };
+
+        let remainder = self.degrees_from_remainder.of(num);
+
+        match wholes.checked_add(&remainder) {
+            Some(res) => Ok(res),
+            None => Err(BaseTypeOverflow {
+                msg: format!(
+                    "Final addition failed on {:?} and {:?} for type {}.",
+                    wholes,
+                    remainder,
+                    type_name::<N>()
+                ),
+            }),
+        }
     }
+
+    // Note that functions like 'add_percents_to' or 'subtract_percents_from' don't need to be
+    // implemented here, even though they are at the 'PurePercentage'. You can substitute them
+    // simply by querying 100 + <your desired addition in percents> or 100 - <your desired
+    // subtraction in percents in the interval (1..=99) >
```

Comment:
Instead of the second sentence, you can directly provide usage examples:

```rust
    // - To increase by X%: use `LoosePercentage::new(100 + X)`
    // - To decrease by X%: use `LoosePercentage::new(100 - X)` where X is in range 1..=99
```